### ISSUE TITLE
add option to override labels / not to apply labels

### ISSUE THREAD

**Issue by giladsh1** [GoogleContainerTools/skaffold#2687]
### Expected behavior
The latest Skaffold release (v0.36.0) has set artifact caching on by default (awesome feature by the way!).
This means I don't need to repeat previous builds when changing / fixing k8s configs.
However, since Skaffold applies custom labels for each run, including the `skaffold.dev/run-id` label, this means that deployments will forced to be configured each run, and they won't be idempotent. 
For example, please look at the following mongoDB deployment (statefullset) - 
```
Name:               mongo-0
Namespace:          riscale-test
Priority:           0
PriorityClassName:  <none>
Node:               gke-gke-dev-europe-we-persistent-pool-43a650d2-3nlv/10.132.0.27
Start Time:         Tue, 20 Aug 2019 16:34:41 +0300
Labels:             app=mongo
                    app.kubernetes.io/managed-by=skaffold-v0.36.0
                    controller-revision-hash=mongo-d4d79d7bc
                    env=test
                    skaffold.dev/builder=google-cloud-build
                    skaffold.dev/cleanup=true
                    skaffold.dev/deployer=kubectl
                    skaffold.dev/profiles=riscale-test
                    **_skaffold.dev/run-id=bfae5d05-c01c-40a8-8516-663dafd4a419_**
                    skaffold.dev/tag-policy=git-commit
                    skaffold.dev/tail=true
                    statefulset.kubernetes.io/pod-name=mongo-0
```
And the following deployment is the exact same, other than the skaffold.dev/run-id tag - 
```
Name:               mongo-0
Namespace:          riscale-test
Priority:           0
PriorityClassName:  <none>
Node:               gke-gke-dev-europe-we-persistent-pool-43a650d2-3nlv/10.132.0.27
Start Time:         Tue, 20 Aug 2019 16:34:41 +0300
Labels:             app=mongo
                    app.kubernetes.io/managed-by=skaffold-v0.36.0
                    controller-revision-hash=mongo-d4d79d7bc
                    env=test
                    skaffold.dev/builder=google-cloud-build
                    skaffold.dev/cleanup=true
                    skaffold.dev/deployer=kubectl
                    skaffold.dev/profiles=riscale-test
                    **_skaffold.dev/run-id=dc524cde-21f0-42dd-aa52-8fcdcf0890c7_**
                    skaffold.dev/tag-policy=git-commit
                    skaffold.dev/tail=true
                    statefulset.kubernetes.io/pod-name=mongo-0
```

I suggest adding a CLI option to turn these custom labels off / override the default Skaffold labels.
  
### Actual behavior
Skaffold applies custom runs for each run, without the ability to remove them.
### Information

- Skaffold version: v0.36.0
- Operating system: MacOS
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v1beta13
kind: Config
build:
  local:
    push: false
  artifacts:
    # web-console
    - image: eu.gcr.io/riscale/web-console
      context: web-console
      docker:
        dockerfile: Dockerfile-dev
        buildArgs:
          ENVIRONMENT: ""

    # backend-services
    - image: eu.gcr.io/riscale/http
      docker:
        buildArgs:
          BUILD_FOLDER: backend/http
          APP_PORT: 2000
    - image: eu.gcr.io/riscale/nats
      docker:
        dockerfile: Dockerfile-additional-copy
        buildArgs:
          BUILD_FOLDER: backend/nats
          APP_PORT: 2000
          SRC_ADD_FILE: backend/nats/*.json
          DST_ADD_FILE: /
    - image: eu.gcr.io/riscale/backend-api
      docker:
        buildArgs:
          BUILD_FOLDER: backend/api
          APP_PORT: 8090

    # db migrations
    - image: eu.gcr.io/riscale/migration-postgres
      context: db/postgres
    - image: eu.gcr.io/riscale/migration-rethinkdb
      context: db/rethinkdb/migrations

profiles:
  - name: test
    build:
      googleCloudBuild:
        projectId: riscale
        timeout: "900s"
        machineType: N1_HIGHCPU_8
    patches:
      - op: replace
        path: /build/artifacts/0/docker/dockerfile
        value: Dockerfile
      - op: replace
        path: /build/artifacts/0/docker/buildArgs/ENVIRONMENT
        value: demo
    deploy:
      kubectl:
        manifests:
          - k8s/gke/riscale-test/dynamic/**
```

### Steps to reproduce the behavior

1. Run `skaffold run -p test`
2. run `skaffold run -p test` one more time
3. all pods will be reconfigured.

**Comment by balopat** [comment]
Thank you @giladsh1 for opening this issue! Can you try with `skaffold run --force=false`? We rely on the runId labels for portforwarding and deployment status checking.

**Comment by giladsh1** [comment]
thank you @balopat for you swift reply!
Running with `--force=false` doesn't solve the issue unfortunately.
I've changed[ this line](https://github.com/GoogleContainerTools/skaffold/blob/master/pkg/skaffold/deploy/labeller.go#L65) to a static string, and compiled locally.
Everything seems to be working properly.
Not sure what other significance these tags have that I'm missing..

**Comment by dgageot** [comment]
I agree with @giladsh1 that it's a regression.
 + We used to be able to run `skaffold run` twice in a row and it would deploy nothing on the second run.
 + Now, the second run is faster, thanks to artifact caching, but, every object is redeployed because of the new `run-id.
I'm going to label this as a P1 bug

**Comment by dgageot** [comment]
See https://github.com/GoogleContainerTools/skaffold/issues/2745

**Comment by dgageot** [comment]
Fixed with #2752
